ui/spells/spells.scss

```diff
@@ -125,6 +125,116 @@
     /* and make rows overlap */
     margin-top: -1px;
   }
+
+  $grid-padding: 2px;
+  $grid-column-gap: 2px;
+
+  @mixin spells-header($grid-template) {
+    display: grid;
+    grid-template-columns: $grid-template;
+    column-gap: $grid-column-gap;
+    background: #93c6f2;
+    padding: $grid-padding;
+
+    > div {
+      text-align: center;
+      margin: auto;
+      font-weight: bold;
+      width: 100%;
+    }
+  }
+
+  @mixin spells-rows($grid-template, $column-count) {
+    display: grid;
+    grid-template-columns: $grid-template;
+    row-gap: 0;
+    column-gap: $grid-column-gap;
+    background: aliceblue;
+    input, input[type=number], textarea, select {
+      width: 100%;
+      margin-left: unset;
+      background-color: white;
+    }
+    input[type=text] {
+      text-align: left;
+    }
+    input, span, select {
+      white-space: nowrap;
+    }
+    span {
+      text-align: center;
+      margin: auto 0;
+    }
+    button, button[type=roll] {
+      margin: 0;
+    }
+    .spell-row.conditional-one {
+      // This makes the notes span all columns
+      grid-column-start: 1;
+      grid-column-end: ($column-count + 1);
+    }
+  }
+
+  @mixin spells-grid($spell-type, $grid-template, $column-count) {
+    .spells-header-grid.#{$spell-type} {
+      @include spells-header($grid-template);
+    }
+
+    div[data-groupname="repeating_#{$spell-type}"] > div {
+      @include spells-rows($grid-template, $column-count);
+    }
+  }
+
+  $arcane-grid-template:
+    // Discipline?
+    minmax(0, 1fr)
```

Comment:
Arcane spells have a lot of columns, and they often extend past their space.
`minmax` allows for them to shrink and grow.
In these cases, we are saying that they can grow up to 1fr.
Example:
![image](https://user-images.githubusercontent.com/5629800/208278535-a7588a7b-60c7-483a-958c-a6c201a661e6.png)